src/Api/ApiServiceProvider.php

```diff
@@ -42,6 +42,32 @@ public function register()
             return $routes;
         });
 
+        $this->app->singleton('flarum.api.floodgates', function () {
+            return [
+                [
+                    'paths' => ['*'],
+                    'methods' => ['GET', 'POST', 'PUT', 'PATCH', 'DELETE'],
+                    'callback' => function ($actor, $request) {
+                        if ($request->getAttribute('bypassFloodgate')) {
+                            return false;
+                        }
+                    }
+                ]
+            ];
```

Comment:
Should this perhaps be class instances instead? I don't particularly like the idea of using array objects for this, to be completely honest.

Reply:
I don't have strong feelings either way, arrays felt simpler to start with.

Reply:
I agree with @datitisev, even though I had mixed feelings. I think it makes more sense to make code strict than vague.